LCM/scripts/python3/PerformRequiredConfigurationChecks.py

```diff
@@ -0,0 +1,115 @@
+
+operation = 'PerformConsistency'
+
+def main():
+    try:
+        run_perform_required_configuration_checks()
+    except SystemExit:
+        exit(exc_info()[1])
+    except Exception:
+        # Python 2.4-2.7 and 2.6-3 recognize different formats for exceptions. This methods works in all versions.
+        formattedExceptionMessage = format_exc()
+        write_omsconfig_host_log('Python exception raised from PerformRequiredConfigurationChecks.py: ' + formattedExceptionMessage, pathToCurrentScript, 'ERROR')
+        raise
+
+def run_perform_required_configuration_checks():
+
+    dsc_sysconfdir = join(helperlib.CONFIG_SYSCONFDIR, helperlib.CONFIG_SYSCONFDIR_DSC)
+    omicli_path = join(helperlib.CONFIG_BINDIR, 'omicli')
+    dsc_host_base_path = helperlib.DSC_HOST_BASE_PATH
+    dsc_host_path = join(dsc_host_base_path, 'bin/dsc_host')
+    dsc_host_output_path = join(dsc_host_base_path, 'output')
+    dsc_host_lock_path = join(dsc_host_base_path, 'dsc_host_lock')
+    dsc_host_switch_path = join(dsc_host_base_path, 'dsc_host_ready')
+
+    if ("omsconfig" in helperlib.DSC_SCRIPT_PATH):
+        write_omsconfig_host_switch_event(pathToCurrentScript, isfile(dsc_host_switch_path))
+
+    if ("omsconfig" in helperlib.DSC_SCRIPT_PATH) and (isfile(dsc_host_switch_path)):
+        use_omsconfig_host = True
+    else:
+        use_omsconfig_host = False
+
+    parameters = []
+    if use_omsconfig_host:
+        parameters.append(dsc_host_path)
+        parameters.append(dsc_host_output_path)
+        parameters.append("PerformRequiredConfigurationChecks")
+        parameters.append("1")
+    else:
+        parameters.append(omicli_path)
+        parameters.append("iv")
+        parameters.append(helperlib.DSC_NAMESPACE)
+        parameters.append("{")
+        parameters.append("MSFT_DSCLocalConfigurationManager")
+        parameters.append("}")
+        parameters.append("PerformRequiredConfigurationChecks")
+        parameters.append("{")
+        parameters.append("Flags")
+        parameters.append("1")
+        parameters.append("}")
+
+    stdout = ''
+    stderr = ''
+
+    if use_omsconfig_host:
+        try:
+            # Open the dsc host lock file. This also creates a file if it does not exist
+            dschostlock_filehandle = None
+            dschostlock_filehandle = open(dsc_host_lock_path, 'w')
+            print("Opened the dsc host lock file at the path '" + dsc_host_lock_path + "'")
+            
+            dschostlock_acquired = False
+
+            # Acquire dsc host file lock
+            for retry in range(60):
+                try:
+                    flock(dschostlock_filehandle, LOCK_EX | LOCK_NB)
+                    write_omsconfig_host_log('dsc_host lock file is acquired by : PerformRequiredConfigurationChecks', pathToCurrentScript)
+                    dschostlock_acquired = True
+                    break
+                except IOError:
```

Comment:
log the exception

Reply:
There already is a log which says was not acquired and that is the only IOError that should be caught.
Plus this runs every 15 mins, we already are facing an issue of log files growing too large for customers so we shouldn't add logs that won't provide any new information.